### PR TITLE
Wrap with pybind11 instead of swig

### PR DESCRIPTION
**Pull request by pschella**

### CODE REVIEW

File: python/lsst/pex/logging/__init__.py [hunk 1]
```diff
@@ -20,5 +20,6 @@
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
+import loggingLib
```
Comment (timj) [inline python/lsst/pex/logging/__init__.py]: are you sure this shouldn't be `from .loggingLib import *`? And why are there two imports here for `loggingLib`?

Comment (timj) [inline python/lsst/pex/logging/__init__.py]: Consider `from __future__ import absolute_import`

File: python/lsst/pex/logging/loggingLib.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
+import lsst.daf.base
+from trace import *
```
Comment (timj) [inline python/lsst/pex/logging/loggingLib.py]: `from __future__ import absolute_import`

File: python/lsst/pex/logging/loggingLib.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+from _threshold import *
+from _logRecord import *
+from _log import *
+from _debug import *
+from _blockTimingLog import *
+from _screenLog import *
+import _common
+
+import lsst.utils
```
Comment (timj) [inline python/lsst/pex/logging/loggingLib.py:4]: Maybe group the external imports together?

File: python/lsst/pex/logging/loggingLib.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    elif isinstance(val, str):
+        return self.addPropertyString(name, val)
+    elif isinstance(val, lsst.daf.base.PropertySet):
+#         return self.addPropertyPropertySetPtr(name, val)
+        raise lsst.pex.exceptions.TypeError("PropertySet type temporarily unsupported")
+    elif isinstance(val, list):
+        for v in val:
+            self.addProperty(name, v)
+    elif isinstance(val, dict):
+        for k in val.keys():
```
Comment (timj) [inline python/lsst/pex/logging/loggingLib.py:71]: Modernize? `for k in val:`

Comment (timj) [inline python/lsst/pex/logging/loggingLib.py:71]: What I like about this is that all the python code is actually in py files and not hidden in .i files. Makes it much easier to futurize.

File: python/lsst/pex/logging/loggingLib.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        self.data().remove(name)
+        for k in val.keys():
+            self.addProperty("%s.%s" % (name, k), val[k])
+    else:
+        raise lsst.pex.exceptions.TypeError("unsupported property type for logging: %s(%s)" % (name, type(val)))
+
+LogRecord.addProperty = _LogRecord_addProperty
+LogRecord.setProperty = _LogRecord_setProperty
+
+# This doesn't appear to be needed anymore (at least not for the unit tests)
```
Comment (timj) [inline python/lsst/pex/logging/loggingLib.py:119]: You'll find out when you build `lsst_apps`.

File: python/lsst/pex/logging/loggingLib.py [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+# tweak Debug
+Debug.default_max_debug = None
+_Debug_wrapped_ctr = Debug.__init__
+
+def _Debug__init__(s, name, maxverb=None):
+    if maxverb is None:  maxverb = Debug.default_max_debug
+    if maxverb is None:  maxverb = -1 * Log.INHERIT_THRESHOLD
+    _Debug_wrapped_ctr(s, name, maxverb)
+
+Debug.__init__ = _Debug__init__
```
Comment (timj) [inline python/lsst/pex/logging/loggingLib.py]: Newline.

File: python/lsst/pex/logging/loggingLib.py (hunk 4 above)
Comment (timj) [inline python/lsst/pex/logging/loggingLib.py:71]: I guess I would understand if the code was left as close as possible to the `.i` version for now so long as we modernize it eventually.

File: python/lsst/pex/logging/trace.py [hunk 7]
```diff
@@ -0,0 +1,5 @@
+from _trace import *
```
Comment (timj) [inline python/lsst/pex/logging/trace.py]: Absolute import. I'll stop mentioning these tweaks from now on.

File: python/lsst/pex/logging/__init__.py (hunk 1 above)
Comment (pschella) [inline python/lsst/pex/logging/__init__.py]: done

File: python/lsst/pex/logging/loggingLib.py (hunk 2 above)
Comment (pschella) [inline python/lsst/pex/logging/loggingLib.py]: done

File: python/lsst/pex/logging/loggingLib.py (hunk 3 above)
Comment (pschella) [inline python/lsst/pex/logging/loggingLib.py:4]: done

File: python/lsst/pex/logging/loggingLib.py (hunk 4 above)
Comment (pschella) [inline python/lsst/pex/logging/loggingLib.py:71]: Indeed, sometimes the python code in the swig files was / is not up to spec. But I decided for clarity to copy it verbatim in most cases. If we indeed go with pybind11 we can clean it up later. If not, then we can still clean it up in the `.i` files with slightly more difficulty.

File: python/lsst/pex/logging/loggingLib.py (hunk 5 above)
Comment (pschella) [inline python/lsst/pex/logging/loggingLib.py:119]: Indeed. Unfortunately it seems like one more possible case of incomplete unit test coverage.

Comment (pschella) [inline python/lsst/pex/logging/loggingLib.py:119]: I don't particularly like leaving uncommented code in like this. But in this case I really think it helps. Part of the final transition to pybind11 (if accepted) should be to go through the Python code and clean it up.

File: python/lsst/pex/logging/loggingLib.py (hunk 6 above)
Comment (pschella) [inline python/lsst/pex/logging/loggingLib.py]: done

File: python/lsst/pex/logging/trace.py (hunk 7 above)
Comment (pschella) [inline python/lsst/pex/logging/trace.py]: Removed file instead. Was not really needed.

**Pull request merged**